src/eventmodel.cpp

```diff
@@ -306,83 +304,77 @@ QVariant EventModel::data(const QModelIndex &index, int role) const
             break;
     }
 
-    int column = index.column();
-    if (role >= BaseRole) {
-        column = role - BaseRole;
-        role = Qt::DisplayRole;
-    }
-
     QVariant var;
-    switch (column) {
```

Comment:
Should this `switch` be combined with the one above? It seems a bit odd to have two when there is no obvious reason to keep both.

Reply:
Guess I could just merge. Kept the original structure to limit the delta a bit, but it shouldn't be that complex change so I'll just do.